network/p2p/inspector/validation/control_message_validation_inspector_test.go

```diff
@@ -0,0 +1,450 @@
+		rpcTracker.On("LastHighestIHaveRPCSize").Return(int64(100)).Maybe()
+		require.NoError(t, inspector.processInspectRPCReq(duplicateMsgIDReq))
+	})
+
+	t.Run("inspectIWantMessages should disseminate invalid control message notification for iWant messages when duplicate message ids exceeds the allowed threshold", func(t *testing.T) {
+		inspector, distributor, rpcTracker, _, _ := inspectorFixture(t)
+		duplicateMsgID := unittest.IdentifierFixture()
+		duplicates := flow.IdentifierList{duplicateMsgID, duplicateMsgID}
+		msgIds := append(duplicates, unittest.IdentifierListFixture(5)...).Strings()
+		duplicateMsgIDIWant := unittest.P2PRPCIWantFixture(msgIds)
+
+		expectedPeerID := peer.ID("peerID987654321")
+		duplicateMsgIDReq, err := NewInspectRPCRequest(expectedPeerID, unittest.P2PRPCFixture(unittest.WithIWants(duplicateMsgIDIWant)))
+		require.NoError(t, err, "failed to get inspect message request")
+
+		checkNotif := checkNotifFunc(t, expectedPeerID, p2pmsg.CtrlMsgIWant, IsIWantDuplicateMsgIDThresholdErr)
+		distributor.On("Distribute", mock.AnythingOfType("*p2p.InvCtrlMsgNotif")).Return(nil).Once().Run(checkNotif)
+		rpcTracker.On("LastHighestIHaveRPCSize").Return(int64(100)).Maybe()
+		rpcTracker.On("WasIHaveRPCSent", mock.AnythingOfType("string")).Return(true).Run(func(args mock.Arguments) {
+			id, ok := args[0].(string)
+			require.True(t, ok)
+			require.Contains(t, msgIds, id)
+		})
+		require.NoError(t, inspector.processInspectRPCReq(duplicateMsgIDReq))
+	})
+
+	t.Run("inspectIWantMessages should disseminate invalid control message notification for iWant messages when cache misses exceeds allowed threshold", func(t *testing.T) {
+		inspector, distributor, rpcTracker, _, _ := inspectorFixture(t)
+		// set high cache miss threshold to ensure we only disseminate notification when it is exceeded
+		inspector.config.IWantRPCInspectionConfig.CacheMissThreshold = .9
+		msgIds := unittest.IdentifierListFixture(100).Strings()
+		expectedPeerID := peer.ID("peerID987654321")
+		inspectMsgReq, err := NewInspectRPCRequest(expectedPeerID, unittest.P2PRPCFixture(unittest.WithIWants(unittest.P2PRPCIWantFixture(msgIds))))
+		require.NoError(t, err, "failed to get inspect message request")
+
+		checkNotif := checkNotifFunc(t, expectedPeerID, p2pmsg.CtrlMsgIWant, IsIWantCacheMissThresholdErr)
+		distributor.On("Distribute", mock.AnythingOfType("*p2p.InvCtrlMsgNotif")).Return(nil).Once().Run(checkNotif)
+		rpcTracker.On("LastHighestIHaveRPCSize").Return(int64(100)).Maybe()
+		// return false each time to eventually force a notification to be disseminated when the cache miss count finally exceeds the 90% threshold
+		rpcTracker.On("WasIHaveRPCSent", mock.AnythingOfType("string")).Return(false).Run(func(args mock.Arguments) {
+			id, ok := args[0].(string)
+			require.True(t, ok)
+			require.Contains(t, msgIds, id)
+		})
+		require.NoError(t, inspector.processInspectRPCReq(inspectMsgReq))
+	})
```

Comment:
Per our previous sync, we also planned to limit the iWant cache misses to the RPCs that contain more than `x` iWants (e.g., `x = 100`). Please add that check to the inspector and support it by a test here. 
https://github.com/onflow/flow-go/blob/7e9a55d3078ae1be8b247fdd32501a31d3dca7bf/network/p2p/inspector/validation/control_message_validation_inspector.go#L331-L333

Reply:
https://github.com/onflow/flow-go/pull/4695/commits/e0e0e64a12a802f80a92251cd8becd1dc100e3a7